### PR TITLE
Allocator: replace balancer IP with pod IPs list

### PR DESCRIPTION
**Pull request by lindlind**

### CODE REVIEW

File: lzy/allocator/src/main/java/ai/lzy/allocator/configs/ServiceConfig.java [hunk 1]
```diff
@@ -18,12 +19,17 @@
...
     private String instanceId;
-    private String address;
+    private List<String> hosts = new ArrayList<>(); // either [<ipv4>] or [<ipv4>,<ipv6>]
+    private Integer port;
     private Duration allocationTimeout;
     private Duration heartbeatTimeout;
     private List<String> serviceClusters = new ArrayList<>();
     private List<String> userClusters = new ArrayList<>();
 
+    public String getAddress() {
```
Comment (imakunin) [inline lzy/allocator/src/main/java/ai/lzy/allocator/configs/ServiceConfig.java:29]: Why last one?

Comment (lindlind) [reply lzy/allocator/src/main/java/ai/lzy/allocator/configs/ServiceConfig.java:29]: ipv6 has a higher priority than ipv4. IPs has [<ipv4>,<ipv6>] format, so we need the last one.

File: deployment/tf/modules/yc/allocator.tf [hunk 2]
```diff
@@ -64,9 +64,19 @@ resource "kubernetes_stateful_set" "allocator" {
             name  = "ALLOCATOR_METRICS_PORT"
             value = local.allocator-metrics-port
           }
+
+          env {
+            name = "ALLOCATOR_HOSTS"
+            value_from {
+              field_ref {
+                field_path = "status.podIPs"
```
Comment (ArtoLord) [inline deployment/tf/modules/yc/allocator.tf:72]: Maybe use service in lzy cluster and ips in datasphere?

**Review by ArtoLord — CHANGES_REQUESTED** [review]

File: lzy/allocator/src/main/java/ai/lzy/allocator/configs/ServiceConfig.java (hunk 1 above)
Comment (imakunin) [reply lzy/allocator/src/main/java/ai/lzy/allocator/configs/ServiceConfig.java:29]: let's do it more explicetly, plz
i.e. let's split it by `,` and parse each part into IPv4 or IPv4 object
and then select on of them (with reason description)

**Review by ArtoLord — APPROVED** [review]

**Pull request merged**